pkg/operator/startupmonitorreadiness/readiness_checks.go

```diff
@@ -95,6 +99,20 @@ func (ch *KubeAPIReadinessChecker) IsReady(ctx context.Context, revision int) (
 		return false, "", "", fmt.Errorf("a node name is required, use the SetNodeName method")
 	}
 
+	if ch.isSNO == nil {
```

Comment:
Isn't the `IsReady` method called only on the SNO platform ?

Worth checking, if it is we could simply remove the code that tries to determine the platform type.

Reply:
`IsReady` being called for all platforms, its part of library-go interface: https://github.com/openshift/library-go/blob/e749b54aff208942299351536cf2bf2e09df2dfb/pkg/operator/staticpod/startupmonitor/cmd.go#L24

(there is a different startup probe for static pods though - https://github.com/vrutkovs/cluster-kube-apiserver-operator/blob/6556d4c60747aad03920ffdffc38bc5ea3132220/pkg/operator/starter.go#L261-L282)

Reply:
I think that `KubeAPIReadinessChecker` is used by the `startup-monitor` and it is run only on the SNO platform.

Check out https://github.com/openshift/cluster-kube-apiserver-operator/blob/ffb092644a35398a8b848522a96f60ac5a7297ef/cmd/cluster-kube-apiserver-operator/main.go#L60


Where is the code that runs this check on all platform?

Reply:
Oh, so in `main.go` we create startupmonitorreadiness, but start it for SNO only in starter.go? Okay, its safe to remove infra check in this PR then

Reply:
Yes. I think that the startup monitor runs only on the SNO platform and as of today it is sole consumer of `KubeAPIReadinessChecker`.